### PR TITLE
helm: Add configuration option for endpoint source IP verification

### LINKED ISSUE #33889: Option to disable SourceIPVerification while chaining CNI
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

higher than v1.15.7 and lower than v1.16.0

### What happened?

While using Cilium in chain with KubeOVN, using Hubble, I detected that Cilium drops packets that are handled by KubeOVN. In particular this packets are generated by the VPC Nat GW, implemented to permit custom VPC - inside KubeOVN - to reach external networks.

Looking around I stumbled upon "SourceIPVerification" that, name intended, should skip the src ip validation and let the packet continue. Enabling that flag, unfortunately, does not work.

### How can we reproduce the issue?

1. Install KubeOVN with Cilium as instructed here: https://kubeovn.github.io/docs/v1.12.x/en/advance/with-cilium/
2. Create VPC
3. Create NatGW
4. ping any external ip such as 1.1.1.1

### Cilium Version

1.15.6

### Kernel Version

5.15.0-116-generic

### Kubernetes Version

1.29.6

### Regression

_No response_

### Sysdump

_No response_

### Relevant log output

_No response_

### Anything else?

_No response_

### Cilium Users Document

- [X] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by CiraciNicolo**
Please ensure your pull request adheres to the following guidelines:

- [ ] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [ ] All code is covered by unit and/or runtime tests where feasible.
- [ ] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [ ] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [ ] Provide a title or release-note blurb suitable for the release notes.
- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [ ] Thanks for contributing!

This PR implements a way to manage SourceIPVerification flag from Cilium's config file.

Fixes: #33889

### CODE REVIEW

**Comment by CiraciNicolo** [comment]
I opened this PR as draft, so we can discuss if the solution is accepted. Therefore we can also implements other config flags that are now hardcoded.

**Comment by CiraciNicolo** [comment]
@squeed any update? 😊

**Comment by squeed** [comment]
@CiraciNicolo apologies, didn't realize you were waiting on me. This seems like the right approach.

(Do note that we make changes to `values.yaml.tmpl` then run `make -C install` to update.

**Comment by SkalaNetworks** [comment]
@CiraciNicolo wanna switch this out of draft?

**Comment by CiraciNicolo** [comment]
@SkalaNetworks working on this right now, will implement other parameters and ask for review

**Comment by joestringer** [comment]
I'd suggest to drop the conntrack local case from this PR. This was deprecated: https://github.com/cilium/cilium/pull/34358 .

Allowing configuration of source IP verification makes sense though, but we should probably add the flag to The other option makes sense though, but it should probably also be specified in the global flags in `daemon/cmd/daemon_main.go`. It would also help to rename the PR to something like "Add configuration flag for Source IP Verification".

**Comment by joestringer** [comment]
Please also drop the merge commits, it's enough to just rebase against the tip of `main` and force-push your updated PR into this branch.

**Comment by joestringer** [comment]
Looking back this took a while to get feedback, probably because (a) it was marked as draft, and generally speaking in Cilium we won't look at drafts unless we're actively collaborating with someone on it, and (b) the PR probably could have been reviewed by anyone in @cilium/sig-datapath , so by marking it as draft and expecting a review from @squeed it meant that if he was busy, on vacation, or so on then it's easy for him to miss that and block the effort. Generally if you don't receive a review within a week or so, I'd encourage you to post a message in the #development channel in Slack to see if someone else can help to unblock it.

**Comment by joestringer** [comment]
Other than that, thanks for the submission :) I think there's some value here and we should be able to help you get this in for an upcoming v1.17 snapshot release :rocket:

**Review by gandro — COMMENTED** [review]
Helm in general looks okay, but there are a few auto-generated files that need to be regenerated. Please run

```
make -C install/kubernetes
make -C Documentation update-helm-values
```

Thanks!

**Comment by gandro** [comment]
Looks good. Could you also clean up your git history so it only contains one or two commits (no merge commits or follow-up fixes, please). Thank you!

**Comment by CiraciNicolo** [comment]
Hi! I just rebased as @joestringer requested; regarding the conntrack parameter, I would be inclined to keep it implemented until the feature is dropped. I assume some folks still use it, therefore it would be something useful for them.

File: install/kubernetes/cilium/values.schema.json [hunk 1]
```diff
@@ -1502,6 +1502,18 @@
             "string"
           ]
         },
+        "enableConntrackLocal": {
+          "type": [
+            "null",
+            "boolean"
+          ]
+        },
```
Comment (joestringer) [inline install/kubernetes/cilium/values.schema.json]: I appreciate the perspective, but I strongly disagree about adding this conntrack local Helm configuration option. The feature is known to be broken in various cases which is why we've deprecated it. We can't add new ways to try to use the feature when we're actively working to remove the feature.
```suggestion
```

**Review by joestringer — CHANGES_REQUESTED** [review]

File: install/kubernetes/cilium/values.schema.json (hunk 1 above)
Comment (CiraciNicolo) [reply install/kubernetes/cilium/values.schema.json]: Didn't know that the feature was broken, sorry about that. I removed the options and updated the documentation.

**Review by gandro — APPROVED** [review]
Helm-wise this still looks good. Please squash your commits into one. Thanks!

**Review by derailed — APPROVED** [review]

**Comment by CiraciNicolo** [comment]
Hi @gandro, commits have been squashed. Do you need a rebase?

**Review by joestringer — APPROVED** [review]

**Comment by joestringer** [comment]
Thanks again for your persistence on working on this @CiraciNicolo, merging :)

**Pull request merged**